flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducer.java

```diff
@@ -326,6 +342,24 @@ private void checkAndPropagateAsyncError() throws Exception {
 		}
 	}
 
+	/**
+	 * If the internal queue of the {@link KinesisProducer} gets too long,
+	 * flush some of the records until we are below the limit again.
+	 * We don't want to flush _all_ records at this point since that would
+	 * break record aggregation.
+	 */
+	private void checkQueueLimit() {
```

Comment:
probably rename it to something different, e.g  `enforceQueueLimit()`? because it clearly does things more than just 'check'

Reply:
I agree.